samples/d3/src/runtime/widget.tsx

```diff
@@ -0,0 +1,79 @@
+/** @jsx jsx */
+import { AllWidgetProps, BaseWidget, jsx } from "jimu-core";
+import * as React from "react";
+import * as d3 from "./lib/d3/d3.min.js";
+
+export default class Widget extends BaseWidget<AllWidgetProps<any>, any> {
```

Comment:
I'll update all other widgets to extends from `React.PureComponent`

Reply:
Thanks, I updated these to use `React.PureComponent` following your example.